ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -713,24 +719,27 @@ public void run() {
         queueEntryCount.set(0);
         metrics.statsRecentEntryQueueSize.update(0);
         startTimeInMs = time.milliseconds();
-        newScanResults = new ScanResults(startTimeInMs, logSegmentForecastOffsetMs, bucketCount, bucketSpanInMs);
+        newScanResults = new ScanResults(startTimeInMs, logSegmentForecastOffsetMs, bucketCount, bucketSpanTimeInMs);
         scanLock.lock();
-        isScanning = true;
-        scanLock.unlock();
+        try {
+          isScanning = true;
+        } finally {
+          scanLock.unlock();
+        }
         Offset firstCheckpoint = index.getCurrentEndOffset();
         ConcurrentNavigableMap<Offset, IndexSegment> indexSegments = index.getIndexSegments();
         Map<StoreKey, Long> deletedKeys = new HashMap<>();
         // process the active index segment based on the firstCheckpoint in case index segment rolled over after the
         // checkpoint is taken
-        if (indexSegments.size() > 0) {
+        if (!cancelled && indexSegments.size() > 0) {
           Map.Entry<Offset, IndexSegment> activeIndexSegmentEntry = indexSegments.floorEntry(firstCheckpoint);
           long indexSegmentStartProcessTime = time.milliseconds();
           List<IndexEntry> activeIndexEntries =
-              getFilteredIndexEntries(activeIndexSegmentEntry.getValue(), firstCheckpoint);
+              getIndexEntriesBeforeOffset(activeIndexSegmentEntry.getValue(), firstCheckpoint);
```

Comment:
No, we do not have such metrics. Would you like me to add these metrics? Currently, there is a total time and and time per index segment for bucketing scan. Together with statsForwardScanEntryCount we can get some idea about the time to first checkpoint and from first to second checkpoint.